core-bundle/src/Resources/contao/dca/tl_page.php

```diff
@@ -257,8 +258,22 @@
 			'exclude'                 => true,
 			'search'                  => true,
 			'inputType'               => 'text',
-			'eval'                    => array('mandatory'=>true, 'rgxp'=>'language', 'maxlength'=>5, 'nospace'=>true, 'doNotCopy'=>true, 'tl_class'=>'w50'),
-			'sql'                     => "varchar(5) NOT NULL default ''"
+			'eval'                    => array('mandatory'=>true, 'maxlength'=>64, 'nospace'=>true, 'decodeEntities'=>true, 'doNotCopy'=>true, 'tl_class'=>'w50'),
+			'sql'                     => "varchar(64) NOT NULL default ''",
+			'save_callback'           => array(
```

Comment:
Should this be a separate listener or is inline fine for this case?

Reply:
I am fine either way, although I tend towards the listener.

Reply:
I think we agreed, that we don’t want to add new methods to DCA classes. Not sure about inline callbacks. cc @fritzmg ?

Reply:
I think we agreed that new callbacks need to be either inline (for really simple ones) or services. However, I think I would prefer a service in this case.

Reply:
Are the DCA callbacks considered to be part of the public API? I think we only [excluded event listeners](https://docs.contao.org/dev/internals/bc-promise/), so we might want to add `@internal` to new methods in DCA files so that we can actually move them later on.

Regarding the discussion where to put things: I don't think this can be generalized. I'm with @leofeyer in principal but could imagine use cases for new independent functionality (or functionality across multiple DCAs) that doesn't fit in there.

Reply:
> Are the DCA callbacks considered to be part of the public API?

No, we already removed legacy `tl_*` methods and even whole classes.

Reply:
They are event listeners. So they are excluded.

Reply:
Changed in a0314b3eb9322cb6f2c6e528247b25e08161e904